### ISSUE TITLE
val_size default calculation for n_examples >50 

### ISSUE THREAD

**Issue by rnyak** [IndicoDataSolutions/finetune#383]
**Describe the bug**
I am fine tuning a GPT-2 model with a train set of 9790 samples. After I trained the model, I checked model configurations and saw that val_size = 0. However based on val_size formula "If n_examples > 50, defaults to max(5, min(100, 0.05 * n_examples))", it should be 100. This was calculated correctly in the previous version of finetune library, but after I updated the library to the latest version I see it is not calculated correctly.

The other issue is the data_size param. What data_size displays ( 'dataset_size': 10459) is not the correct number of my training set size (it should be 9790).  I do not understand how this number is calculated. Please see the train and test set sizes below.

**Minimum Reproducible Example**
`trainX.shape, testX.shape, trainY.shape, testY.shape
((9790,), (4196,), (9790,), (4196,)) `

#model training command:

`model = Classifier(n_epochs=2, base_model=GPT2Model, tensorboard_folder ='/workspace/checkpoints', max_length= 1024)
model.fit(trainX, trainY)`

The output of `model.config`
{'grid_searchable': {'n_epochs': [1, 2, 3, 4],
  'l2_reg': [0.0, 0.1, 0.01, 0.001],
  'lr': [0.000625, 6.25e-05, 6.25e-06]},
 'low_memory_mode': False,
 'interpolate_pos_embed': False,
 'save_adam_vars': True,
 'shuffle_buffer_size': 100,
 'dataset_size': 10459,
 'batch_size': 2,
 'visible_gpus': None,
 'n_epochs': 2,
 'seed': 42,
 'max_length': 1024,
 'weight_stddev': 0.02,
 'save_dtype': None,
 'val_set': None,
 'per_process_gpu_memory_fraction': None,
 'adapter_size': None,
 'embed_p_drop': 0.1,
 'attn_p_drop': 0.1,
 'resid_p_drop': 0.1,
 'clf_p_drop': 0.1,
 'l2_reg': 0.001,
 'vector_l2': False,
 'autosave_path': None,
 'keep_best_model': False,
 'early_stopping_steps': None,
 'min_secs_between_eval': 60,
 'eval_acc': False,
 'val_size': 0,
...}

**Expected behavior**
The val_size  should be calculated from the given formula "If n_examples > 50, defaults to max(5, min(100, 0.05 * n_examples))".

**Additional context**

**Comment by madisonmay** [comment]
Hi @rnyak,

Thanks for raising an issue -- this is a case of out of date documentation.  In the 0.8 release we opted to move to setting `val_size=0` by default and running validation on an opt-in basis only (we typically finetune for a fixed number of epochs rather than monitoring val loss and early stopping). The previous behavior can be enabled by passing in `val_size='auto'` to the Classifier.

With regards to dataset size, when examples exceed the maximum length of 1024 tokens they are split into overlapping chunks and processed on a per chunk basis.  It seems like this happened for ~5% of your examples in the dataset which has increased the overall dataset size. To disable this behavior you can pass the flag `chunk_long_sequences=False` to the `Classifier` constructor.  

I've got some time scheduled to give our docs some TLC this next week, will make sure the `val_size` docs get updated to reflect the current behavior and make the behavior of the `chunk_long_sequences` argument more explicit, but thanks again for bringing these bugs to our attention!

**Comment by rnyak** [comment]
Thanks for your quick response. That helps.

**Comment by madisonmay** [comment]
Docs have been updated to be a bit more clear on the points you brought up.  Thanks again for the issue!

**Comment by rnyak** [comment]
Is the max val_size calculated from the given formula "If n_examples > 50, defaults to max(5, min(100, 0.05 * n_examples))"? In other words, if I have 30000 samples can I set the val_size = 3000 or 5000? When I do that, I do not see any loss values, for train and val set, on TensorBoard.
